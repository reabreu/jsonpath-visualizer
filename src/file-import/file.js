import React from "react";

export const File = ({ onChange, hasError }) => {
  return (
    <div>
      <label htmlFor="file">Choose a JSON file:</label>
      <input
        onChange={onChange}
        type="file"
        id="file"
        name="json"
        accept=".json"
        role="button"
      />
      {hasError && <p>Oops, looks like there was an error, please retry.</p>}
    </div>
  );
};
