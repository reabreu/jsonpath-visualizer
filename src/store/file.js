import { decorate, observable, computed, action } from "mobx";
import jp from "jsonpath";

class JSONP {
  data = {};
  query = "";

  setData(data) {
    this.data = data;
  }

  setQuery(query) {
    this.query = query;
  }

  get matchSet() {
    console.log("match set computed");
    const matchSet = new Set();

    if (this.query !== "") {
      try {
        const paths = jp.paths(this.data, this.query);

        paths.forEach((element) => {
          matchSet.add(jp.stringify(element));
        });
      } catch {
        console.log("Invalid expression analyzed");
      }
    }

    return matchSet;
  }
}

export const JSONPState = decorate(JSONP, {
  data: observable,
  query: observable,
  matchSet: computed,
  setData: action,
  setQuery: action,
});
