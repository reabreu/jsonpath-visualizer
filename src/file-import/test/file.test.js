import React from "react";
import { render, fireEvent, wait } from "@testing-library/react";
import { File } from "../file";

describe("<Search /> spec", () => {
  it("Contains file input and calls onChange", async () => {
    const onChangeHandler = jest.fn();

    const { asFragment, getByRole } = render(
      <File onChange={onChangeHandler} />
    );

    expect(asFragment()).toMatchSnapshot();

    const input = getByRole("button");

    fireEvent.change(input);

    await wait(() => expect(onChangeHandler).toHaveBeenCalled(), {
      timeout: 300,
    });
  });
});
