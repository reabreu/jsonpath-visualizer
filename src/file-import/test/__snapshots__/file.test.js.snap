// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> spec Contains file input and calls onChange 1`] = `
<DocumentFragment>
  <div>
    <label
      for="file"
    >
      Choose a JSON file:
    </label>
    <input
      accept=".json"
      id="file"
      name="json"
      role="button"
      type="file"
    />
  </div>
</DocumentFragment>
`;
